building/trusted_applications.rst

```diff
@@ -373,35 +373,44 @@ third step the binary and its signature are stitched together into the full TA.
 Offline Signing of TAs
 ======================
 
-The TA dev kit does sign an application as last step of the linking process. For
-example, the file ``ta/arch/arm/link.mk`` in the :ref:`optee_os` source tree
-contains the statement
+There are two types of TAs that can be signed offline: the in-tree TAs, which come
+with the OPTEE OS (for example the ``pkcs11`` TA) and any external TA coming from the user. Offline signing
+runs directly on `.stripped.elf` TA files and there is a slight difference as how they are
+generated. The in-tree `.stripped.elf` TA files are generated during the OPTEE OS compilation, while
+with external TAs it is usually done by the user. In both cases however, the signing process is the
+same.
+
+Offline signing is done with the following sequence of steps:
+
+0. (Preparation) Generate a 2048 bit RSA key for signing in a secure, offline
+environment and extract the public key.
```

Comment:
I was expecting an example command for the key generation (such as an OpenSSL command). I know that most people following this procedure would probably use an HSM with specific tools and arguments, but having a basic working example is still valuable to understand what the process expects.

Reply:
Just pushed a new commit.